File: ver-ios/src/budg/budgEffect.swift
import gb
import UIKit

func budgCopyResult(_ result: String) {
    UIPasteboard.general.string = result
}

func budgPasteMorningBalance() {
    let txt = UIPasteboard.general.string ?? ""
    budgSet(F.pastedMorningBalance, txt)
}

func budgPasteSpent() {
    let txt = UIPasteboard.general.string ?? ""
    budgSet(F.pastedSpent, txt)
}

File: components/budg/android/budgEffect.kt
package org.opengamestudio

import android.content.ClipboardManager
import android.content.ClipData
import android.content.Context

fun budgCopyResult(
    ctx: Context,
    result: String
) {
    val clip = ctx.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
    val item = ClipData.newPlainText("Result", result)
    clip.setPrimaryClip(item)
}

fun budgPasteMorningBalance(ctx: Context) {
    val txt = clipboardText(ctx)
    budgSet(F.pastedMorningBalance, txt)
}

fun budgPasteSpent(ctx: Context) {
    val txt = clipboardText(ctx)
    budgSet(F.pastedSpent, txt)
}
